gammapy/region/core.py

```diff
@@ -0,0 +1,231 @@
+
+        Parameters
+        ----------
+        pixcoord : tuple
+            The position or positions to check, as a tuple of scalars or
+            arrays. In future this could also be a `PixCoord` instance.
+        """
+        raise NotImplementedError("")
+
+    def area(self):
+        """
+        Returns the area of the region as a `~astropy.units.Quantity`.
+        """
+        raise NotImplementedError("")
+
+    def to_sky(self, wcs, mode='local', tolerance=None):
+        """
+        Returns a region defined in sky coordinates.
+
+        Parameters
+        ----------
+
+        wcs : `~astropy.wcs.WCS` instance
+            The world coordinate system transformation to assume
+
+        mode : str
+            Convering to sky coordinates can be done with various degrees of
+            approximation, which can be set with this option. Possible values
+            are:
+
+            * `'local'`: assume that the field of view is small and that
+              pixels are square, so that e.g. a circle in sky coordinates
+              would be a circle in pixel coordinates. This is the fastest and
+              most commonly used for e.g. photometry.
+
+            * `'affine'`: approximate any deviations from the 'local'
+              assumption by an affine transformation, e.g. a circle would
+              become a rotated ellipse.
+
+            * `'full'`: return an arbitrarily complex polygon in sky
+              coordinates that represents the full level of distortion due to
+              the conversion from pixel to world coordinates. The degree of
+              exactness can be controlled by the ``tolerance`` argument.
+
+        tolerance : `~astropy.units.Quantity`
+            The tolerance for the ``'full'`` mode described above.
+        """
+        raise NotImplementedError("")
+
+    def to_mask(self, mode='center'):
+        """
+        Returns a mask for the aperture.
+
+        Parameters
+        ----------
+        mode : { 'center' | 'any' | 'all' | 'exact' }
+            The following modes are available:
+                * ``'center'``: returns 1 for pixels where the center is in
+                  the region, and 0 otherwise.
+                * ``'any'``: returns 1 for pixels where any of the pixel is
+                  in the region, and 0 otherwise.
+                * ``'all'``: returns 1 for pixels that are completely inside
+                  the region, 0 otherwise.
+                * ``'exact'``: returns a value between 0 and 1 giving the
+                  fractional level of overlap of the pixel with the region.
+
+        Returns
+        -------
+        mask : `~numpy.ndarray`
+            A mask indicating whether each pixel is contained in the region.
+        slice_x, slice_y : `slice`
+            Slices for x and y which can be used on an array to extract the
+            same region as the mask.
+        """
+        raise NotImplementedError("")
+
+    def to_shapely(self):
```

Comment:
Again, we probably won't support shapely, so I'd say remove for now.
(they do have some nice stuff, e.g. for overlap testing, but if we really want that we can always add it back)